source/common/rds/basic/config_traits_impl.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/rds/config_traits.h"
+
+namespace Envoy {
+namespace Rds {
+namespace Basic {
```

Comment:
What is "Basic"? Seems a bit non-descript.

Reply:
Yes, I'm in a little trouble with the naming related to these templates.
I'm open to every suggestion.

Reply:
`Common`?

Reply:
How about `Manager`? To make the things more clear we could rename the `Rds::RoutConfigProviderManager` to something like `Rds::RoutConfigProviderRegistry`. 
Or `Generic`?
If not, I rather vote for `Common`, these templates are not subject to be base class.

Reply:
Renamed to Common